platforms/pf-robot-motion-control/motion_control.cpp

```diff
@@ -434,6 +434,13 @@ void pf_motion_control_reset(void)
 void pf_disable_motion_control()
 {
     pf_motion_control_platform_engine.disable();
+
+    // Small wait to ensure engine is disabled
+    ztimer_sleep(ZTIMER_MSEC, motion_control_thread_period_ms);
```

Comment:
Maybe it can be more safe to introduce an internal semaphore inside `enable/disable` methods instead of a timer ?

This is an area for improvement, probably not necessary now.

Reply:
true. i'll check this as we have time to do it.

Reply:
done